### DISCUSSION TITLE
Clarification about KH support

### DISCUSSION THREAD

**Discussion by willlunniss** [nathanmarlor/foxess_modbus#432] [Support] [answered]
Hi,

Looking to order a system including KH10 and cube batteries and wanted to double check what is and isn't automatable.

The wiki currently says that the KH only supports reading basic values https://github.com/nathanmarlor/foxess_modbus/wiki/Supported-Features. However, looking through #171, the code and recent discussions (i.e. #425) it looks like everything is supported except for Read/write charge periods and that is just because the implementation does not support it yet, rather than it not being possible?

Key things I want to be able to achieve are:
* When in a cheap off-peak window (when Intelligent Octopus is charging my EV so this can happen at anytime in the day) either switch the battery to force charge from grid or to at least to not discharge the battery (will depend on import/export rates as to which would be preferred)

* On days where I could exceed my kW export limit switch the work mode to Feed-in First for a period of time to ensure that I leave room in my battery to charge with any excess above normal self use + export limit

I believe both of those should be possible, but would appreciate confirmation?

Also if anything else is left to be implemented for the KH I would be more than happy to do the investigations and contribute back if we go ahead.

Thanks

**Comment by canton7** [comment, marked as answer]
Oops you're right, that table was out of date. I've just fixed it. And yep, it looks like charge periods can be supported also.

**Reply by willlunniss** [reply]
Great, thank you for confirming.

**Reply by Mwalton4uk** [reply]
> Oops you're right, that table was out of date. I've just fixed it. And yep, it looks like charge periods can be supported also.

The table still says charge periods aren't supported for KH :/
